### ISSUE TITLE
gcc7 compilation error

### ISSUE THREAD

**Issue by tokumeiwokiboushimasu** [inolen/redream#178]
Hi, I use fedora26 with gcc-7.2.1.
Latest commit cannot compile.

/home/hoge/aho/redream/src/guest/gdrom/disc.c: In function 'disc_create':
/home/hoge/aho/redream/src/guest/gdrom/disc.c:265:46: error: '%s' directive output may be truncated writing up to 128 bytes into a region of size between 127 and 255 [-Werror=format-truncation=]
   snprintf(disc->uid, sizeof(disc->uid), "%s %s %s %s", disc->product_name,
                                              ^~
/home/hoge/aho/redream/src/guest/gdrom/disc.c:265:3: note: 'snprintf' output between 4 and 516 bytes into a destination of size 256
   snprintf(disc->uid, sizeof(disc->uid), "%s %s %s %s", disc->product_name,
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            disc->product_number, disc->product_version, disc->media_config);
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [CMakeFiles/redream.dir/build.make:615: CMakeFiles/redream.dir/src/guest/gdrom/disc.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:152: CMakeFiles/redream.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

**Comment by s1nka** [comment]
Some problem. Try:
`cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_C_FLAGS=-Werror=format-truncation=0 ..`

**Comment by tokumeiwokiboushimasu** [comment]
It works. Thanks.

**Comment by s1nka** [comment]
But it's not solution. Only hacks. I'm not sure that emulator work perfectly.

**Comment by inolen** [comment]
I don't have a gcc7 installed, could someone test to see if this is fixed as of fca17ca8f2d4a4dbc79caada274c08b65f372afb.

**Comment by s1nka** [comment]
Build perfectly. Thanks.
